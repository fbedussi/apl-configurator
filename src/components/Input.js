const Input = ({ labelText, name, minLenght, required, type }) => {
    var t = type ? type : 'text';

    if (required) {
        labelText += '*';
    }

    return <div className="form-field">
            <label className="form-label" htmlFor={name}>{labelText}</label>
            <input className="form-input" id={name} name={name} type={t} minLength={minLenght} required={Boolean(required)}/>
    </div>;
};

export default Input;
