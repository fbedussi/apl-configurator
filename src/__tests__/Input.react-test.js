import React from 'react';
import Input from '../components/Input';
import renderer from 'react-test-renderer';

const props = {
  labelText: 'label text',
  name: 'company',
  required: true,
  minLength: 2
};

test('Input component', () => {
  const component = renderer.create(
    <Input {...props} />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});