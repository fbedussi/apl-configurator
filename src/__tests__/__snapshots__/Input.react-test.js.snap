// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component 1`] = `
<div
  className="form-field"
>
  <label
    className="form-label"
    htmlFor="company"
  >
    label text*
  </label>
  <input
    className="form-input"
    id="company"
    minLength={undefined}
    name="company"
    required={true}
    type="text"
  />
</div>
`;
